core/ledger/kvledger/kv_ledger.go

```diff
@@ -887,14 +893,64 @@ func (l *kvLedger) GetMissingPvtDataTracker() (ledger.MissingPvtDataTracker, err
 	return l, nil
 }
 
+type commitNotifier struct {
+	dataChannel chan *ledger.CommitNotification
+	doneChannel <-chan struct{}
+}
+
 // CommitNotificationsChannel returns a read-only channel on which ledger sends a `CommitNotification`
 // when a block is committed. The CommitNotification contains entries for the transactions from the committed block,
 // which are not malformed, carry a legitimate TxID, and in addition, are not marked as a duplicate transaction.
 // The consumer can close the 'done' channel to signal that the notifications are no longer needed. This will cause the
 // CommitNotifications channel to close. There is expected to be only one consumer at a time. The function returns error
 // if already a CommitNotification channel is active.
 func (l *kvLedger) CommitNotificationsChannel(done <-chan struct{}) (<-chan *ledger.CommitNotification, error) {
-	return nil, errors.New("yet-to-implement")
+	l.commitNotifierLock.Lock()
+	defer l.commitNotifierLock.Unlock()
+
+	if l.commitNotifier != nil {
+		return nil, errors.New("only one commit notifications channel is allowed at a time")
+	}
+
+	l.commitNotifier = &commitNotifier{
+		dataChannel: make(chan *ledger.CommitNotification, 10),
```

Comment:
Can you explain why you chose `10` for the buffer size? I don't have an issue with it but I'm wondering where it came from.

Reply:
No special calculation. Just to give enough buffer if at times gateway becomes slow in consuming the events, the commits are not blocked.